nexusd/main.go

```diff
@@ -50,42 +50,100 @@ func main() {
 	// Create router and realms from config.
 	r, err := nexus.NewRouter(&conf.Router, logger)
 	if err != nil {
-		fmt.Fprintln(os.Stderr, err)
+		logger.Print(err)
 		os.Exit(1)
 	}
 
-	// Create a new websocket server with the router.
-	s := nexus.NewWebsocketServer(r)
+	// Create and run servers.
+	var closers []io.Closer
+	if conf.WebSocket.Address != "" {
+		// Create a new websocket server with the router.
+		wss := nexus.NewWebsocketServer(r)
+		var closer io.Closer
+		if conf.WebSocket.CertFile == "" || conf.WebSocket.KeyFile == "" {
+			closer, err = wss.ListenAndServe(conf.WebSocket.Address)
+		} else {
+			// Config has cert_file and key_file, so do TLS.
+			closer, err = wss.ListenAndServeTLS(conf.WebSocket.Address, nil,
+				conf.WebSocket.CertFile, conf.WebSocket.KeyFile)
+		}
+		if err != nil {
+			logger.Print(err)
+			os.Exit(1)
+		}
+		closers = append(closers, closer)
+		logMsg := fmt.Sprintf("Listening for websocket connections on ws://%s/", conf.WebSocket.Address)
+		logger.Printf(logMsg)
+		if conf.LogPath != "" {
+			// Print to console if not already.
+			fmt.Println(logMsg)
+		}
```

Comment:
seems that these are not right, should all go to logs or to console IMO.

same below.

Reply:
done